src/App.tsx

```diff
@@ -15,58 +15,74 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export const App: React.FC = () => {
+  const [choosenProduct, setChoosenProduct] = useState<string>(`Jam`);
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  const onClickDeselect = () => {
+    setChoosenProduct('');
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      {choosenProduct.length ? (
+        <h1 className="title is-flex is-align-items-center">
+          {choosenProduct} is selected
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={onClickDeselect}
+          />
+        </h1>
+      ) : (
+        <h1 className="title">No goods selected</h1>
+      )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good, i) => {
+            const isGoodChoosed = choosenProduct === good;
+            const onClickSelectProduct = () => {
+              setChoosenProduct(good);
+            };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            return (
+              <tr
+                data-cy="Good"
+                key={i}
```

Comment:
In this case index as key won't cause any problems but its better not to use it at all. Always look for unique value from the values that you map.